packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart

```diff
@@ -240,20 +248,23 @@ class _DraggableSheetExtent {
   _DraggableSheetExtent({
     required this.minExtent,
     required this.maxExtent,
+    required this.snapPoints,
     required this.initialExtent,
     required VoidCallback listener,
-  }) : assert(minExtent != null),
-       assert(maxExtent != null),
-       assert(initialExtent != null),
-       assert(minExtent >= 0),
-       assert(maxExtent <= 1),
-       assert(minExtent <= initialExtent),
-       assert(initialExtent <= maxExtent),
-       _currentExtent = ValueNotifier<double>(initialExtent)..addListener(listener),
-       availablePixels = double.infinity;
+  })  : assert(minExtent != null),
+        assert(maxExtent != null),
+        assert(initialExtent != null),
+        assert(minExtent >= 0),
+        assert(maxExtent <= 1),
+        assert(minExtent <= initialExtent),
+        assert(initialExtent <= maxExtent),
+        _currentExtent = ValueNotifier<double>(initialExtent)
+          ..addListener(listener),
+        availablePixels = double.infinity;
 
   final double minExtent;
   final double maxExtent;
+  final Set<double> snapPoints;
```

Comment:
This would be a natural place to use a set (order doesn't matter, duplicates don't make any conceptual sense). However, I hope to next tackle: https://github.com/flutter/flutter/issues/45009.

As a part of that PR I'd like to create a `Future<void> animateToSnapPoint(int index)` function where users can programmatically animate to a given snap point based on it's index in `snapPoints`. This functionality necessitates a list.